### PR TITLE
Ensure we still read/refresh remote state when activate is set to false.

### LINKED ISSUE #27258: Unable to use 'for' expression with TypeSets with computed attributes
## Edit

This looks to be related to https://github.com/hashicorp/terraform/issues/26419 which, although closed as being 'fixed' by a 0.14.x release of terraform, is still an issue it seems.

### Terraform Version

```
Terraform v0.14.2
+ provider registry.terraform.io/fastly/fastly v0.21.1
```

### Terraform Configuration Files

```terraform
// https://registry.terraform.io/providers/fastly/fastly/latest/docs
terraform {
  required_providers {
    fastly = {
      source  = "fastly/fastly"
      version = "0.21.1"
    }
  }
}

provider "fastly" {
  api_key = "<REDACTED>"
}

variable "mydict" {
  type = object({ name = string, items = map(string) })
  default = {
    name = "My Dictionary"
    items = {
      key1 : "value1"
      key2 : "value2"
    }
  }
}

resource "fastly_service_v1" "demo" {
  name = "teratest"

  domain {
    name    = "teratest.global.ssl.fastly.net"
    comment = "test"
  }

  backend {
    address          = "127.0.0.1"
    name             = "test-backend1"
    port             = 80
    auto_loadbalance = false
  }

  dictionary {
    name = var.mydict.name
  }

  activate = false
}

resource "fastly_service_dictionary_items_v1" "items" {
  service_id    = fastly_service_v1.demo.id
  dictionary_id = { for s in fastly_service_v1.demo.dictionary : s.name => s.dictionary_id }[var.mydict.name]

  items = var.mydict.items

  lifecycle {
    ignore_changes = [items, ]
  }
}
```

### Debug Output

I already REDACTED the API key, but just to be on the safe side I've decided to encrypt the trace output with the Hashicorp public key and exported (below) the ascii armored version.

<details><summary>ASCII ARMORED OUTPUT</summary>
<p>

```
-----BEGIN PGP MESSAGE-----

hQEMA9X0O2Qy6nfzAQf9GHXZLir2NHsHuwF2uvrg+2zZ1YVXy8m9OWvgVqj1rM78
SJNGRFfY8SsI0f0ZhF4mEzWPWCQKoZhyFSSsHquhFN00ZQ3VXR8pLZorY14TX719
yD4QahbfEuSXdsOQH7n+IPOayOm1dHkg5G+uC3NgsQMJTfS95UoFv4TE8qkht+rr
d/yX+W6zzX9+nFQ1nISBeMsmuf1WUjUxvwn+Hkeft7HVyCmUqD24p1TFsgsNz+6B
PTQU8+X4AcCkJ0xTAKKmexOtEE81ni1ybCblRwrTQ9MEOZhmkGJ7jlO5UMaPulSw
1yGuG2+HLdQk9pVXbyb6CWnQnjbfco0Pc7HFY0cgf9LsAVjw7QOevzbRBAlYD5Tn
7NOyCP2bPNmjDAg0Hq9ss0kC2VEmazZWGQ92XrTmcAlzW2eFlzmjsiAoyEdeIItX
z6oMfEXvGjsMdCPYrZxoTtVid3yuBmUQCmDLO92PgfeYcWYd1+NSgRjGJZh8ulWi
3hJedVO2YGNXwxlNtv1FG3Sjil9RQnZOY3SB9qOU3pFMIzxLz4iyB2NrbkAGhugc
9Mx7ZV+Uqb+rFCveKthHiu/dY3bCh+k1OV8jFymRdXe0x4syukuZPIqGKGFcJjDS
sPUJ8NeoKV7ovJS/5Q58P0A7JJ9PmuL2R4OntsqqxP9htPzmgWfEgcnGhgQjezgi
6jwK7t+yk72aRTUfS5eduFFiCJWpN4RzwlTYNh14C8ppN+vbUKl0yJlRA+CErNrs
JnCettjqXHnJ4m2OkXkRO9sOu8VhqVkNeLF3cQ/YGJKF8Ji5TwB1sKggyUgLhsih
pMAxnTBPY/LAxbudkxuaO6uZ/ocXllAIc6pAi1W4k67OguvD8LLGgJ+/2IHEQKcW
DnsoNZO7r3+XCP+IwaVKKiGimCH4xTsQ4V2+6aXzA4MPQMdFL39pGWoyqVyd587e
iXMQ3ui4LhyGWYxRre542oEeXbQ4qNgYiQIpXAHoUo/YQWxa7HGAJYyj7VwqFwVL
+eHlb0WwAtlXxN57C2WRdtuOliFt9Kku01ewLvYcz+hqXFM1+e9QanAjAXCLYuF/
vWcyap7b6isx+J2uOVK3tlK2vOh9QerpBZ/8y50NfQ4sgFbjeWduPJcXVkGj4res
Zdz9AYCvv7TBc5T4pcxyU2TT7oO+03qSOfCZFq8A+FcdJ3bpRGlLvMLJeWzxNmAQ
fdUHrz7B2yx9Qwrzs5THrhJSW6Q/LS/Jn7kkOABFl5JF4llrrXBDldHHVk6s2huB
NSVy+4Wq5RvB8FyV1kowQj+jc34RiFFomtFok7EZXFIeq4+gMiWuGgoCYuar//kO
AOO/2kbaoTmn4STqpw0RICXjjc6VnLUSsFhvAYsM3/RhtzEk0XvGAtl8KT3+sqW7
bDuDyV4ZeD7zfPIHms4EL7tGcihoqADoUw5LVg1RS9BGx7hD/zDF7WQ0ZlagsO6g
N/xH4z67GOfbHgDNjn5wZL5SenPvRy6DFVqR2Wu8dOkwgh62DIFBQqxvCCXQkyre
4je9orjmbfudTwjqH5NO8joRNYCYrHLOOLtzkcG5DWfpH9Gr8+uuvg98Undi1eQx
ZPXixcBU2JuH1odRmtsCCDnf81GgQuPkkM14+3H+eVd3thGagcymnCquXD/vMzM8
NXMB0+aT1tZsG1BqmJp0ppqAEQ7MqB5DUmHTHbP1f6lf0S46Lw5dH/uOazmkrKq0
La2ILClS26ZMu9WjNNHPhVqnkpKg/+D0Dktda6YyWHCa3pZdH6D6gm/dANdE6Fel
kLNNMVqTdtH3+l0GN0K+IZzmCZzkzLy222cswfVY4PGaV5FlfzWv5VYyaXv1m70+
R1YkOyxeBKKNP8grpcBTdYnPaa9+4fnqs0UKw/BQTWdrSm6ocUmc8cnr7hnE4FJX
boBbLQe8putryuN4nG9aA/jGF/UVpS8kwTi6UAq1yWuWLoKsiceBQrbYDuMxqT4y
lKevy2K+IFtw2St02M+5O44sMBuPLGby8Bl4AhrzeIENzVk2SaYzVxPoD4U6KU0R
aWV2kRSL89L48zFi34V87C76YNIUYJNcsWbvMmA9AwVMrjuK4rWPrwS5czQ5KrG7
nPP/BGjIYHpbdmZjHcLlHD2Q4fcvxhkdZ/dpmYonQ3Wr/cAhsPzyq6U4Q5hioILT
ipWzRXSefqUk+aa2SUxSG5VoNw949CY+VJQqpdA5C4JTOSyepv/d+Zl2pN5+M7QS
mGZFMPhiiWHY4tLjwkJQIUvCHUo8grrAWF6wIG5NCgK2S2B1AgrwanGZWMjGRS5e
kZZ20yBQZBvPCjJcNRYSW8IvtRhkGEj4sC1ONizxjIpMY4V7MFlo6kVCzKZyPj5l
a7hZ2nI8Td4wDLUS8Uk+MkXMJ577FxH249lievKSbJyjJUQaKktVjCJvItxnurjp
fZT5Lr4XaTxDcO1Dzml1m9zEo3k0TpAmtrr1AvbleagxP45++x9Dz/ZcqPdw86w1
Gga7PMZtg7FRQ6uVIICMKuIkp3Zlm91TvOt2Q05p8umhgRFULYPD4s5NdK8BO2/W
pop75XMHDxOZYvQ2BD+H3hJWy7PJzKii+uyIRdRSr1oLcmNHJPk+D4drQkFQlt9o
w/0t2BPoPb8NbIDrf/HpyootzVCw8mijVbH8B8YkpGa3Tb9tNbMKHWhXzpJX2omc
XCJ+mylgJsfa6AsNgBl06DVEE/YWooYf7jTquY6x/jx8+k1odR5xalOfZGQHwryG
OoiwBo586bzkPLCTITr3cj4V/6Ngq5je0jeuxhcwIi5MyWdjU74isS0cC3QF6aNR
pUbKVEx2pANHIsznyjUWUr4m+c1gSsUi6bAIcN1yEmWBaVsDZi3LOHFiw+st8o9c
apRBAEEE04n2H58LjDODXX3tNFlUILyujTYJcCT/4ZZWpiKvNrA88s573DOE499q
52ctlGQO/uG8xOqtmlckpRXsNMA6ndsDHT2Z2413/aqBLYf0tMrEzDOtHdmWgrg2
stXQRvcv1BdNsdE1bDFbgQqOHOhHMhUTEkDt4mpOwy+N8t3ewm3bQaT97zdbpwGa
Wv+oPz8ufaj5RucNaClXnPn+nf+H2pvlVFWlkUsRezIkhVLpgidXmNinRWYVg7MJ
2hgTdfb2xqJAIwkFyfGky6Pv9b5N3YeX7iOR2QLH5Pt2/ULAxQU2/b+dWW8V4jXP
Avc/f3eMwo5NKY7dncF5rUUPwXkfTNKb+oEgftPJj8DPL71asoznFbJ9yUS+gwJl
0deKVobchcGr6wcW/vROdhbo9DoyzW8IM8ZHMwgXae/SJdSMFMsAfF6ew9jCjUW7
zDryW+mZSLKDUPgNQRjf0gZXQjg9ZTUIpSqqZBInV3fhIfFeTpFZSN3Aq2p46WzS
YHXSaH3ocoBg4jyWyM81gw3sKfWhdzcHhRG7NYRygVl1ZzzKC+kl4b/JbX46Y2D/
lfY1Vgu7lv6htZb8kGhSPmskT+ErfLZ2itpFtDYEs1henslOfRYhmyMAP+zcMRpF
ICbvI9s6dq1LiXHo4Q8d/xwRwzDUPQYK/r007ect786bAXkeL5+jZfTKZM6EALxE
WhOr8/GFaLnEBISwIXYYhDVX0CyThJ/UK9bWj+9S+DNRyArCTROfo9hRONvqaNZb
YQb23Go5Oktg3UWEhi6aWXJ8Hnk4tOROvtXhyj/ihVHXT5TXcGqxOkGJwq+ChvFy
E7k5Wev8TneEpri3+eK2V4/WXbZmbObMJJbJI9V/HKA/dIVnMZuOV4Dwh73rtbRK
+dCM6/NVZ7UHB5XYBc5hc9Kl6xXElHHEYxtKLCWZY5b+to8D/t1MzoqFja9gMcvF
n0gpVzqXKz83Tky2oVD2Z5NKuIVdSDd13nOcjRhVL7GQbDOnXprqXS+kY0I1Q3Rf
Qwmnl8Ak1ucxtT9aJWjpmxY8ymlRFL9EgDHvrwR/7jnw0u7AJnFFS+zhvcpmmYm3
X8KciPdhLF+8IBZfbtQqTBYgoQgczFosAFq7nvQU3T6b6YhE8tF79wAMEpF/nPm/
3QUUdJbsReistrIQqXp6ipdXhW9zuk++9q1OmW5u8z2DlCmxTBfVK9ZDWyfC42Te
R7RkhPEKE6WQi7OibdaLmuJ3i7JPM5wKdhH4QSldVtbEplZvmGJaBI4WaXMMjRvB
aKgEH6PJjeCDwUDp3Zeql3Mm67q6+68HtrmbKT+45WywAEK1GiaGKsqnqN5fh7K1
HNte2RQqR8eXQL6XCJR8I7AeYGN8AY6JmYkEiFUgp/uGuJVCnScAv0QOYwBtTH9o
GFkNNpme77JI3B7Px5NCGC5adHNj3LY7ACSvM8so6pDQ8U/Uj9SgSE3iIhUvAQsS
Gf6yoJtqf9TXTfBub2aYUtEFesflzQrtfUK9FCmblnQyAEWuMbDq4hturHOgCa8a
2RWAOl6pTS/7XlkCvzH/XxChpm62lPQ5QXN7llqquhiSHf75yIc8iA4Rs9WtFrQO
ZkyfMvnA5e7t47vBNhq0M5+qUUZ+AzQA0ijG7JDQsx6dVAbo2NP/BzzWpvj7RXp2
hr72BRcYIGxK3aXfbC48rSDUFwiM2SLL3fej9yDDhCEgsXt24SpGdw0XwAUbnzj1
mCziOHhWyCONrHHC3/16sC/qqahQT1GEh6s8gWMrLYVPQdoJ+Lfim/qVZshEVW4P
zsg9FDgRRC/lQkW7FgerifFJkITGKhfuiTltQ9FUedpDNaVxxTIB7z4VHzpLHGT7
6h5PzqkhUYxOo1eAdIPbEscoLBymlIdMYllgbN1meFkqW75uV0wtM4s1RBts/zr6
Bz+m72lrqQIeUiT7MS3jKIyZm1ky93TL+bd4SZnDLQLavzF45IRGPoOdnCUsXPNV
v/nikwpR6z32K/682C8kJfEN//lvkAycIarXkxhGusPzZ3kCSn41EE6buXo7wbxS
xo9bwN+rxEMZ4vvwIwtfss/WsW/qUMzyszO/oogVlRFdedete+wy0R7NHWmvhwMS
gKDN3ATZgZupASjr4viCsUjwi0Uk6KAhGL7bYzj5mEXv4rqzC2MbS9H9cpbIVAF+
Q+SSOkkrJwEwqCJXs672QQ029WNzfc7NE4/vaBDedfUx93L69+alNjBuIhfJmrg4
ZfFddiYt6wQX/W/63tvXWxBlSjMtg/tazPveUMx4mxlf88mxtbCMdGjCw642OBDZ
RW2OoLXJYC/hZ0xlgO7K+vVKbPZi6OtwNnHyyIPj/sHOlfM7Zi5zE0yz1u2glSfa
NsnFTsmK8i4noHfw+Q78r2JlZ3Xg/t14r52u1RjOtJmU6YAoPH8gfm87t6dVVQ4h
nup5dbYmIUfpdmlfQXSW4YCqJRHDEb7Dx405uNZQLDkT200qh1b9tUzsnkvR7obP
IBYtKDvTsN26isMplkYaSKHh3wCmYIV7XSg1pCx+cwzVXSvBXrvX2yC3xScGYkB6
XDJj21bfSNhhWRO+1XKtqEVtxk17DskznggZFWoL8jldN0J8ARdBxh7WgOTIcotM
eFfuDr+0pRJCAOEKQ6lp3qFwL1eOyIjasdzu55lKOktJfEmtprC1QdncRQGA0beN
HOcfOJkTXbAOdGn076M16yZAuRQJMZ7p+idH2nLRu1Djf8IOZt/I6A5X8mW73l+4
X4aoHJeCDRaUnkE1Ok/U+N/+cpjEBjUgxmLx3CzTJ2lOkjHLRnB8yEw2M0dObNJr
miScJ4oZZuBK5z5/+PfNksh1tVOLWpqvHmcCN028KCADgtvO4Do4tmR0y6GY+l1o
lfTE+wLjA/IKDUTrKJlVAYQZmLZ95nSj6rpwW0EM/RXbxu2MYmmD/AF6CJaF6iB4
Fuv1aAivXhT3hwkNN4KNLSzUt78IaZJ+JIH2nAHa4RysmCPxlfNbjOPBoaWZ+J7a
789HjpuYA8bjYd2PKJwPKvcTN19ssXbKM5GZr5qud3PDTujKTfvc8CsrNb81r9Ex
60gP0Em7Laebl4pf/JVU3LV0mQN3n+GSM4qBFZTBBkdkWi3sklAbjaQza/hgOfvi
3TOSnXJXOazpvu4+FvlRlput75sDhbl96Qe8WknGg3/H/nBKR61gXSZHAsHvLyLG
7KcsIbogCiX69wp47Jn1qP1Cfx8GFzKhyO3MchJInqi+DB/enqJKVlgg4h10lThG
nUF025CwTK0BASDuTK9HCXAsHSz+WWUuNn6SWnZsNIVru8swuxG/qgEz+xKAVlW6
KPb64V2K3gALSPS4eHo4TSU8lrZGXbsENnlFyv7lA39nc7nxCn/QjsKuNRGaAPgp
ztyq7ex3uiZ43R7E/1Z8BPOHDqukWRZiV78+t08GxmT1iAXzMBUYouz8QDrZeIDl
o8t+XW5oR/1Vq5FyKErYXn8xZaRMOHUk9gV7t5UoTusXie+ZODvo+/aDIhpWpADH
Ens1KtQzRMk/w6DSDTS7gxJQ406TaavCBDXYI9QJWPQKSgBwUqOT8a47WBUXU/Qg
FGg0cVaF9Kc3OypIMQMv93cyGpdMW5G01LGfPFgaxAB84VeTrbxgxkFiKMI6kmya
/9BOLJGDbQhZR5iyGUWSHC/TwovS2JijZ8YmiEopMb9ctZNrpsIAnW8zfUEaKjqU
k+10lgiuvgivIzYrAdAtim97KwgVFO4lNybyIf0WeyZF0JX2YpM3XhJc5cJtHVlR
cJYleY2FW9vEmpQGyOmfDHHwsG1pCp6zMLhak+9+xZLYRtK7ZP3glrMDnYozGFWp
7FL+APxdHfk6xw5Eh27hVSfF/8P1kR3J53mTpOSZcW+Rlug8VawQSUxDTDhm0GrB
Uv1mXZDT2X6RQs8MI9rvcEJ8kfwuYj2fLD9rowUdg6z6qfIudAbV+VNzf+0BBgRz
U/msXC1jG1xilQi7+hryVNKuPNiOt57P6ecf9C3l8F+nMDPR4nw84Vz2ReUfGobY
MGL4UwNA6T7bO6EiQ/ETc4eSVxB/sRT+xDDiGBUDNfbEU692y0L2iWi1NoGj1dPQ
Ev93Unjp6/93nMar9F/T4mcTKh+8uzKevUn9xRV9SMvkQuM3+5r4sUBq8y4ACnwT
cnXHnUjxyBpxWwVBYo2ffQT+tH/Jp5wYpWouVP6uwTzN7iCkPv0kSSshTvAXTJ/6
kWb0vPFuJqrxmTdXGtAgDl5p1zQDZLwcYd1ElIPjohdH1uUMb0l8cObqDE4SXsMG
be2NSrduwt6kwphsGKjUdsoIYsQ/J5alC3dmxD38LEjnmLriVSJ94mwKl+Lu9xeB
DUOMx/Y7LLn/4TTvJ3C/JgwxDLx2t1GcwP/VLr4q4mus4RNfBb2A9TLVuCsbkABO
JQfPF6fKoR7x++AKn7OjecvzqVNa9Qs/7eUoxDfQEZ+15ePPvEsYrcXE6RGgUU5t
beJvAymSVI3YbdYpK+MHwu+Ng1JgeU8AC9nQYyrTANnbKWiydG7Dvp9CFsQFHu1B
EV1whx8XrlPDHyeT8yranWSTAgG6uje9GztW3QNT4XtKBGlJIzcpoRICIQmSAPYH
xd2yENUv1fgbLNlW8v3jdsi8CjEJPcm3/B3QLxyRCMbzus1egNPRvgEq0u+Jxk9U
1OAw5kPp6gYDhiXb/ivawyQq8lKO7tLIuBwjpcJgwdHlacBzdXyWVr/3xkNETbwn
ED39+gTNtOL5PM2WqAzQODY+k7M6e6M39NhQY1Y51JHcB58+xWDdsdOplfLqt0Pf
7SyYuJWz6bIdtzyI+LRBygWQP9tW4c0FmjsHMZN0nEwDcY7KrIntHsX8GxS+iHHu
9RXVC/NwcR1pny7FEelMZ8ujkUwPoC++UrdKmiMR50N3y+y7HgR7W0yOt0djQwQR
qycmgwwldbqrwBF4QFUtdvDYE+dzcPpzTGQSf0ksIxNzw4S1ByeEVJwS1T7xRCDB
T/5Y531zq98TaTizu5dd1eJZo0f9MEwBktnxAZ02vwW4FS5lfPNgyq7U0h2FStIB
vuvde3dDob1fno6PM9lYaDHTyOGpES674IVU6dKa+L/57x+xwIgCp9n5VAz+Gv0F
avkxR70iVH8zmUOdG4I8AntaQ6+ZK3aNU072mYljfJB73LJv0MepdgplKZvZ8Hpw
qnqyDtXd9eRX4Jje9VljS9s5CKffO5OxAPcI7x/sSYBa8VeQt8k8+hpf9mqz9ktK
qVgh4pOWOiyUGlqAnDVjc5eSFKVuat2KaNWqZQNrvKPoOvEiP9cz2urri/3noz8E
fv5Q4Vrf+Q6hucJ+kQpKFgGMEqYae/7PylVjQAWuXFj00hkrpkIDjTd/WQ8m2Hzu
X5yUuIkMAuOqZch6/QcwtvrF2oRPSQCaGI7mDz+PVfKCh96il5cGSaDYpFadbOIm
cOx5g6O1OSqRr874W0XHJRan6RqRMvKXhSMV16y7HWMDsvs8pMJTcldUMoKGi5Ty
m3d8wPylxgf8ARVSYAlCzjjr7trup9HEaQTMOxcaVzuIE6pYDaNRj0k+taHUw3jR
C4RdfeQpQkfMQY9QcoFxtao+OaWpVIRpLvZVjzIK4/8dC6M+xvEHuZRaGanZgLXs
7hhvb/Bvx7Xud9x/OJrd6BqgDMiq0P6JqZfvTaTm4ZWn7uo7GMK5SSZLsTBn/9Mm
3wZBiyndn27jRjlQV4Vp8bmFAHhgnrKAkXG0OSHIEFd3ktJzm3Iy8RVhIAVGlfzd
nZU0ikOReq8pSrScwhOl79ktm39yeCHnCgjBJO4tpC1R5abaxyicTBxS9Q24AHbt
WNOEUaGYfEiTuFbFTVTBuuZWQMYvSEpucvx23zY8dSU8gIO7tJDKPjj3XSqsJhs+
qQoEYRWb2NviPL63fkTK9gdTzcuzunYvB34ZmSuCXgq3K4I4bwpNx/id1xUfOCti
LzceBiID7YQw7nuHPgKSg6vKRNfW688ep30CvVMB0sAcGDejyvlNqHaMKusg4+xQ
+CoqgfQbtCDahZK3hlnRDxG0sWRdZAuOUQ1S6WfPr5D7SgR+TTZpjYQk2cSvitQ0
u4T8kHRBAMdvY0t8Jvt4OX6wsUGF9tTs1+D+GxuqKSLmFkXZqqyKVq9YtPf+ZUCm
7dlf9Hz8MEsPGnqQOUZU3D2fks3U10cTf8vg0xTbQL9YfCgq2q2xQAiXl3hWxUFR
2eYaGGU0DqU/1SpSeEZC+iYI8E+XpfRa65O3opbzsr37RInqxCgCFgyHunt1Tid4
zljKV9hEuoNz0oc4UPtMo14qYrXNEZ1qajNufAcGp3favkSRaPiMTPs4uDlz0kEr
xTCsbNKBPNK276lxmlId2cdhJW+KjSG98qsxv4Xx3OnTQvXR7I/cNoxI3GKMlsQo
oJPgf99kB51zEHU7JrH+cX/egv04Imbj4spFEleNNhkbSkjtiNkFF0d+Vw4UMa/4
cx4noXISM1VG80Y+lZIODfLEzLvwgH0h4WKfi/yFh9uyuowglh13j6436jPq5qJB
l/S8XeqDpBek0wpj4tP+Jzg4S3Nm1M+5yGbaPxwAIkLs19ktxpN8JDDVwvXfTSqU
/87hJ4aqG3e+wVUtLSe8a5/SjOurNCfl9tziOxp8EdfT2ONrn6KmyEZdXRBwSmy2
m8JYgjFQ3fcdHi/DWBFzEa1KKeKtjxYzbnVwT0hqzxhSMPoqEDKnBRvO7XYY4XVo
YzDzuHCsmPI+3uSn5RwSIMNZSa7FXiJEHS+1CWun2YvisHmVc+Zm5fwnEX8xQHC2
gsJrkTe7oG4tNnOO31hXI98pQeKHC5D5MccUHJc2gMbN24Y3TQ7Cn8s172Ryf+MJ
fbLbJ5aSnG3kI3pm8g9RPXT8wH9A+fBsC5jky4na2GGOrlytcjasxlnM2RcyTf7j
dntItVQrkO7b07KIqZIpweAN0nAiirffeaDieSv/luY5ATbFk/QUSqruRa9Nf/NV
uDbnnsdvZMlZV3TcDYMyOPSkhZcAKT4uHWZSb8tdGx1//69OoUUfUMUb8nhhHwQY
v+2w6fD07+Jkv+tGvbo6QhPMiirvVBg3gb/dSjm5/AMs+ku9xotxQH62YNC1zRj8
6DybIppfcPi7Pdrw1x704SJ15RTSrmjtt/VlOlJ+swXtssi/0PHN+yhRAXzzkuvH
rhnOw1WpPWXcDZlW1pLYfOn21UlTJ5+ZTPRyy4loiuwIlGMH3YOEoVKgBs5fR3YC
0/j51lFQK+Vdy7tJkTJVhBk/9xlGpCMnhwdIrMGbY8FLqF8ofmPA67dBxtKeEqt1
O26VBJKqWDbhKNoxdSvwt4Eka8F2qZUXRZfgNa5SsgeE9VQdsszP++Mj8xXPV34a
5e5Zx/Q6U/f1ixepb+4HRYnZ0z905gefFwVDLMiqlKExm6RRelCzu+lgverTV4gx
PmTqZyEjynsW6eX3EZVUAInNzm94RoewvD5YiR5D8KMO0SuYwS/6o5KX+Y+VDTqg
xunA0aPDIxaVONi7Tj1vFhY+V6K0PWPPAA4DIC1ZqHIfR3Cv0x+wYVMVVUSxNs8q
RWRqRRJhaPfVHFR65gj1wcMoJyByOC+FxtQRIEgAqgE+xf6chSC9006KKdXGxgxL
0F/lKa2ujBvbPC3tMzXgiBJLnqEbf2cBmW8Vk5whJvy86ukA+ts+9wFjfJlgDT8Q
6IJ6tRxDdtjtuzTNqIUtY+N4yvnronbN2+bLqOrXXDjGR3PIwO+dpGAmN57k/7uv
1jYTuXteYYKvaF081jiLX6GVzkhG05gRwRu6326nGh7uU1xBzrqKv10zCA1Ccn8u
xybNSCqGBWJkhoc1syYGtY0TO1ezy2FazrQCRYVXlWgleuvms6muNcorOfF1azuv
D6guvt7R7uahQa7QZB99lWY1cRcLsXgbnuFe+Q/cQmT8WwU4iPALHqIT/9VrE+wt
/zZSYzQNwMHK85sq9uxMykeB3HaV2/xirQx8+jJ1fNS+DiFMNAooeWoQ54CrLIOz
B3xfsF2vPB6s5PmQfAbd523MHI4MpjwkylUlAIvXxGPCOf6heFSAIuJgIAs1foEr
Y+zyO58UgreEyRsacldj03QAIqNj5VNDkyYFz/+RSITjTZCdq9dDVHbgy5JRod1m
66/4P200EzCDiGbP1nEE8BqUsGhJraz+wmMSVo8EiPZ9m5O0JHmBxQyJ0Kex16Xn
rEJh+pOS3zRG4lJb9zdKAwwgnc9jGxytPsBnTTIRpbw+64X1gRWwVgJquWgdWOJi
42vvB+L4QSbq/8bZjc0rukgnY2SmqG3ODH6BasUZz0D/y6Oxew==
=IsLf
-----END PGP MESSAGE-----
```

</p>
</details>

### Expected Behavior

The dictionary items should have been added to the dictionary.

### Actual Behavior

Running `terraform apply` shows the following error.

```
fastly_service_v1.demo: Refreshing state... [id=4Ow7CNZGupBaWAocDaVWte]
fastly_service_dictionary_items_v1.items: Creating...

Error: Error creating dictionary items: service 4Ow7CNZGupBaWAocDaVWte, dictionary , missing required field 'Dictionary'

  on test.tf line 78, in resource "fastly_service_dictionary_items_v1" "items":
  78: resource "fastly_service_dictionary_items_v1" "items" {
```

This error only occurs on initial apply, all subsequent applies are fine once the service resource has first been provisioned.

> See similar issue: https://github.com/hashicorp/terraform/issues/26419

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`

### Additional Context

I've tried three different ways of defining the items and each one fails...

1.

```terraform
resource "fastly_service_dictionary_items_v1" "items" {
  service_id = fastly_service_v1.demo.id

  dictionary_id = lookup({
    for d in fastly_service_v1.demo.dictionary : d.name => d.dictionary_id if d.name == var.mydict.name
  }, var.mydict.name, "")

  items = var.mydict.items

  lifecycle {
    ignore_changes = [items, ]
  }
}
```

2.

```terraform
resource "fastly_service_dictionary_items_v1" "items" {
  service_id = fastly_service_v1.demo.id

  for_each = {
    for d in fastly_service_v1.demo.dictionary : d.name => d if d.name == var.mydict.name
  }

  dictionary_id = each.value.dictionary_id

  items = var.mydict.items

  lifecycle {
    ignore_changes = [items, ]
  }
}
```

3.

```terraform
resource "fastly_service_dictionary_items_v1" "items" {
  service_id    = fastly_service_v1.demo.id
  dictionary_id = { for s in fastly_service_v1.demo.dictionary : s.name => s.dictionary_id }[var.mydict.name]

  items = var.mydict.items

  lifecycle {
    ignore_changes = [items, ]
  }
}
```

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
Fixes an edge case in the `fastly_service_*` resources, in which we don't refresh remote state if the `activate` field is set to `false`. This was introduced in https://github.com/fastly/terraform-provider-fastly/pull/269, when we refactored the service resources to the "block" model and introduced the base service definition interface.

### What?
After a couple of bug reports related to `activate = false` via Fastly support, we investigated further and found that if a configuration has the `activate` field set to false AND it has no previous active version the state wasn't being read. This manifested itself as  a runtime error in certain situations, such as another resource referencing the state in its configuration. 

It turns out we were assuming that the service will always have an active version number greater than 0 within the  `Read` method and gating the read on that condition. 

Fixes: https://github.com/hashicorp/terraform/issues/27258

### Note:
I've ran the tests I introduced in this PR to confirm this fixes the  issue. However still ned to run the full suite before merging, which I'll try and do in the coming days.

### CODE REVIEW

File: fastly/base_fastly_service_v1.go [hunk 1]
```diff
@@ -450,7 +450,12 @@ func resourceServiceRead(d *schema.ResourceData, meta interface{}, serviceDef Se
 	d.Set("version_comment", s.Version.Comment)
 	d.Set("active_version", s.ActiveVersion.Number)
 
-	if s.ActiveVersion.Number == 0 && isImport {
+	// If we are importing or `activate` is set to false, temporarily set the
+	// service.ActiveVersion number to the latest version supplied via the get
+	// service version details call. This is to ensure we still read all of the
+	// state below.
+	shouldActivate := d.Get("activate").(bool)
+	if s.ActiveVersion.Number == 0 && isImport || !shouldActivate {
```
Comment (phamann) [inline fastly/base_fastly_service_v1.go]: I'm on  the fence if this this is the cleanest way of solving this, or instead changing the gate on read conditional logic on line 165. Thoughts?

Comment (Integralist) [reply fastly/base_fastly_service_v1.go]: Let's talk through the alternative implementation first before we go ahead and approve/merge this changeset.

**Review by Integralist — APPROVED** [review]

**Comment by phamann** [comment]
As discussed with @Integralist separately, this is now ready to merge and full acceptance test suite is passing (other than some unrelated WAF ones which are broken due to external API changes).
